src/perf/isolates.md

```diff
@@ -0,0 +1,286 @@
+
+All Dart code runs in [isolates]({{site.dart-site}}/language/concurrency), which
+are similar to threads, but differ in that isolates have their own isolated
+memory. They do not share state in any way, and can only communicate by
+messaging. By default, Flutter apps do all of their work on a single isolate –
+the main isolate. In most cases, this model allows for simpler programming and
+is fast enough that the application's UI doesn’t become unresponsive.
+
+Sometimes, though, applications need to perform exceptionally large computations
+that can cause “UI jank” (jerky motion). If your app is experiencing jank for
+this reason, you can move these computations to a helper isolate; allowing the
+underlying runtime environment to run the computation concurrently with the main
+UI isolate's work and takes advantage of multi-core devices.
+
+Each isolate has its own memory and its own event loop. The event loop processes
+events in the order that they’re added to an event queue. On the main isolate,
+these events can be anything from handling a user tapping in the UI, to
+executing a function, to painting a frame on the screen. The following figure
+shows an example event queue with 4 events waiting to be processed.
+
+![The main isolate diagram]({{site.url}}/assets/images/docs/development/concurrency/basics-main-isolate.png){:width="50%"}
+
+For smooth rendering, Flutter adds a “paint frame” event to the event queue 60
+times per second (for a 60Hz device). If these events aren’t processed on time,
+the application experiences UI jank, or worse, become unresponsive altogether.
+
+![Event jank diagram]({{site.url}}/assets/images/docs/development/concurrency/event-jank.png){:width="50%"}
+
+Whenever a process can’t be completed in a frame gap, or the time between two
+frames, it's a good idea to offload the work to another isolate to ensure that
+the main isolate can produce 60 frames per second. When you spawn an isolate in
+Dart, it can process the work concurrently with the main isolate, without
+blocking it.
+
+You can read more about how isolates and the event loop work in Dart on
+the [concurrency page][] of the Dart
+documentation.
+
+[concurrency page]: {{site.dart-site}}/language/concurrency
+
+<iframe width="560" height="315" src="{{site.youtube-site}}/embed/vl_AaCgudcY" 
+ frameborder="0" allow="accelerometer; autoplay;encrypted-media; gyroscope; picture-in-picture" 
+ allowfullscreen></iframe>
+
+## Message passing between isolates
+
+Dart’s isolates are an implementation of the [Actor model][]. They can only
+communicate with each other by message passing. When messages are “passed”
+between each other, they are generally copied from the sending isolate to the
+receiving isolate. This means that any value passed to an isolate, even if
+mutated on that isolate, doesn’t change the value on the original isolate.
+
+The only objects that aren’t copied when passed to an isolate are immutable
```

Comment:
OK, this is a global change - I won't keep calling it out from here on. Change the double and single quotes to the straight up and down kind.  :)

```suggestion
The only objects that aren't copied when passed to an isolate are immutable
```